src/Fantomas.Tests/IfThenElseTests.fs

```diff
@@ -947,4 +944,128 @@ type TransferAmount(valueToSend: decimal, balanceAtTheMomentOfSending: decimal)
             invalidArg "valueToSend" "Amount has to be above zero"
 """
 
-       
\ No newline at end of file
+[<Test>]
+let ``keep then in same line as if with comments, 825`` () =
+
+    let c = { config with
+               KeepIfThenInSameLine = true
+               PageWidth = 80
+               MaxIfThenElseShortWidth = 0 }
+
+    formatSourceString false """type TransferAmount(valueToSend: decimal, balanceAtTheMomentOfSending: decimal) =
+    do
+        // comment
+        if balanceAtTheMomentOfSending < valueToSend then // comment
+            invalidArg "balanceAtTheMomentOfSending" // comment
+                "some very very long error message" // comment
+        if valueToSend <= 0m then // comment
+            invalidArg "valueToSend" "Amount has to be above zero" // comment
+"""  c
+    |> prepend newline
+    |> should equal """
+type TransferAmount(valueToSend: decimal, balanceAtTheMomentOfSending: decimal) =
+    do
+        // comment
+        if balanceAtTheMomentOfSending < valueToSend then // comment
+            invalidArg "balanceAtTheMomentOfSending"  // comment
+                "some very very long error message" // comment
+        if valueToSend <= 0m then // comment
+            invalidArg "valueToSend" "Amount has to be above zero" // comment
+"""
+
+[<Test>]
+let ``keep then in same line as if, else if, else, 825`` () =
+
+    let c = { config with
```

Comment:
Could you move all these tests to a new file KeepIfThenInSameLineTests.
There we can override the config once and keep the test DRY.
Similar to what I did in [NewlineBetweenTypeDefinitionAndMembersTests](https://github.com/fsprojects/fantomas/blob/master/src/Fantomas.Tests/NewlineBetweenTypeDefinitionAndMembersTests.fs#L7)